Gems/EMotionFX/Code/EMotionFX/Pipeline/RCExt/Actor/ActorBuilder.cpp

```diff
@@ -366,7 +366,45 @@ namespace EMotionFX
                 auto mesh = azrtti_cast<const SceneDataTypes::IMeshData*>(it->second);
                 if (mesh)
                 {
-                    outMeshIndices.push_back(nodeIndex);
+                    outMeshIndices.emplace_back(nodeIndex);
+
+                    // Don't need to add mesh node except it is a parent of other joint / mesh node.
+                    // Example:
+                    // joint_1
+                    //   |____transform
+                    //   |____mesh_1 (keep)
+                    //         |____transform
+                    //         |____mesh_2 (remove)
+                    //         |      |____transform
+                    //         |_______joint_2
+                    //                |____transform
+                    // emfx doesn't need to contain the "end-point" mesh node because mesh buffers ultimately stored in a single atom mesh.
+                    // NOTE: Joint and mesh node often has a transform node as the children. To correctly detect whether a mesh node have a joint
```

Comment:
nit: Joint and mesh node often have a transform node as their children. To correctly detect whether a mesh node has a joint....

Reply:
fixed